tests/test_settings.py

```diff
@@ -816,6 +817,30 @@ class Config:
     assert Settings().dict() == {'foo': 'foo_secret_value_str'}
 
 
+@pytest.mark.skipif(sys.platform != 'linux', reason='Linux tests')
+def test_secrets_uppercase_filename(tmp_path):
+    (tmp_path / 'SECRET_VAR').write_text('foo_env_value_str')
+
+    class Settings(BaseSettings):
+        SECRET_VAR: str
```

Comment:
This test doesn't have `case_sensitive = True` - therefore `secret_var` shouldn't be populated.

You'll need two tests one with `case_sensitive = True` and one without showing the functionality is not correct.